### PR TITLE
make dependencies not released be able to cache

### PR DESCRIPTION
**Pull request by xiaohuilam**
This pull request is coming to enhance issue #617 

Testing snapshots:

image
image

picture 1 is a laravel project,
picture 2 is felixfbecker.php-intellisense it self, and does not contain dev dependency.

### CODE REVIEW

File: src/Indexer.php [hunk 1]
```diff
@@ -151,8 +151,13 @@ public function index(): Promise
                     // Check if package name matches and version is absolute
                     // Dynamic constraints are not cached, because they can change every time
                     $packageVersion = ltrim($package->version, 'v');
-                    if ($package->name === $packageName && strpos($packageVersion, 'dev') === false) {
+                    if ($package->name === $packageName) {
                         $packageKey = $packageName . ':' . $packageVersion;
+                        if (strpos($packageVersion, 'dev') !== false) {
+                            $packageKey = $packageName . ':' . (isset($package->source) && isset($package->source->reference) ? 
+                                $package->source->reference : (isset($package->dist) && isset($package->dist->reference) ? 
+                                    $package->source->reference : $packageVersion));
```
Comment (felixfbecker) [inline src/Indexer.php]: I think a normal `if`/`else` would be easier to read than a nested ternary.

Could there be any case where none of these are set?

**Review by felixfbecker — COMMENTED** [review]
LGTM

**Comment by felixfbecker** [comment]
@xiaohuilam do you want to get this merged?

**Comment by smuuf** [comment]
This would be so, so awesome.

**Comment by driskell** [comment]
What was needed here?

**Comment by driskell** [comment]
Hi @xiaohuilam
Just wondering what the status was before closure as curious to see if I could help. Thanks

**Comment by xiaohuilam** [comment]
> Hi @xiaohuilam
> Just wondering what the status was before closure as curious to see if I could help. Thanks

This is a PR out of date, so i just turned it closed

**Pull request closed without merging**